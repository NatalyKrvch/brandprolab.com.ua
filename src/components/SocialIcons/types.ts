export type SocialIcon = {
  _key: string;
  _type: 'socialLink';
  url: string;
  label: string;
  whiteIcon: {
    _type: 'image';
    asset: {
      _ref: string;
      _type: 'reference';
    };
  };
  colorIcon: {
    _type: 'image';
    asset: {
      _ref: string;
      _type: 'reference';
    };
  };
};

export interface SocialIconsProps {
  socialLinks: SocialIcon[];
  isColored?: boolean;
  className?: string;
}
