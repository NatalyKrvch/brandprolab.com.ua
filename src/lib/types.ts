export type Size = 's' | 'm' | 'l';
export type Screen = 'mobile' | 'tablet' | 'desktop';
