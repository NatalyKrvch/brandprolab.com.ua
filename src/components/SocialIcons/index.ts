export { default as SocialIcons } from './SocialIcons';
export * from './types';
