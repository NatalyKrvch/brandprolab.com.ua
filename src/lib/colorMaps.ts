import {
  MainSocialCardColor,
  socialMediaColorClassMapType,
} from '@/components/Cards/SocialMediaCard/types';

export const socialMediaColorClassMap: socialMediaColorClassMapType = {
  [MainSocialCardColor.RED]: {
    bg: 'bg-red-default',
    text: 'text-red-default',
  },
  [MainSocialCardColor.BLUE]: {
    bg: 'bg-blue-default',
    text: 'text-blue-default',
  },
};
