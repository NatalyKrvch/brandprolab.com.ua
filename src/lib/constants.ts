export const ALL_DATA_QUERY = `
{
  "header": *[_type == "header"][0],
  "hero": *[_type == "hero"][0],
  "cases": *[_type == "cases"][0],
  "services": *[_type == "services"][0],
  "testimonials": *[_type == "testimonialsSection"][0],
  "subscribe": *[_type == "subscribeSection"][0],
  "video": *[_type == "videoSection"][0],
  "expertise": *[_type == "expertiseSection"][0],
  "diagnostic": *[_type == "diagnosticSection"][0],
  "footer": *[_type == "footerSection"][0]
}
`;

export const YOUTUBE_REGEX =
  /(?:youtu\.be\/|youtube\.com\/(?:.*[?&]v=|embed\/|v\/|shorts\/|live\/))([^&?/]+)/;

export enum IconClassType {
  SERVICES = 'services',
  EXPERTISE = 'expertise',
  SOCIAL_MEDIA = 'social-media',
  SOCIAL_MEDIA_MAIN = 'social-media-main',
}
export const REVALIDATE = 60;
export const ENDLESS_CAROUSEL_SPEED = 50;
export const SLIDER_RIGHT_SHIFT = 2;
export const SLIDER_LEFT_SHIFT = -1;
export const AMOUNT_OF_TESTIMONIALS_WORDS = 15;
export const AMOUNT_OF_CASES_WORDS_DESKTOP = 13;
export const AMOUNT_OF_CASES_WORDS_TABLET = 9;
export const AMOUNT_OF_CASES_WORDS_MOBILE = 13;
