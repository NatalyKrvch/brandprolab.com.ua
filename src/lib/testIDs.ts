export const INTERNAL_LINK_TEST_ID = 'internal-link-test-id';
export const EXTERNAL_LINK_TEST_ID = 'external-link-test-id';
export const FULL_WIDTH_CONTAINER_TEST_ID = 'full-width-container-test-id';
export const LIMITED_WIDTH_CONTAINER_TEST_ID =
  'limited-width-container-test-id';
export const LEFT_LIMITED_CONTAINER_TEST_ID = 'left-limited-container-test-id';
export const ICON_COMPONENT_TEST_ID = 'icon-component-test-id';
export const ICON_COMPONENT_IMAGE_TEST_ID = 'icon-component-image-test-id';
export const LABEL_COMPONENT_TEST_ID = 'label-component-test-id';
export const BUTTON_TEST_ID = 'button-test-id';
export const UNDERLINED_TEXT_TEST_ID = 'underlined-text-test-id';
export const USER_BADGE_TEST_ID = 'user-badge-test-id';
export const USER_BADGE_IMAGE_TEST_ID = 'user-badge-image-test-id';
export const USER_BADGE_NAME_TEST_ID = 'user-badge-name-test-id';
export const TESTIMONIAL_CARD_TEST_ID = 'testimonials-card-test-id';
export const TESTIMONIAL_TEXT_TEST_ID = 'testimonials-text-test-id';
export const LIST_TEST_ID = 'list-test-id';
export const LIST_ITEM_TEST_ID = 'list-item-test-id';
export const SIMPLE_CARD_TEST_ID = 'simple-card-test-id';
export const DIPLOMA_CARD_TEST_ID = 'diploma-card-test-id';
export const DIPLOMA_IMAGE_TEST_ID = 'diploma-image-test-id';
export const DIPLOMA_OVERLAY_TEST_ID = 'diploma-overlay-test-id';
export const PHOTO_CARD_IMAGE_TEST_ID = 'photo-card-image-test-id';
export const PHOTO_CARD_TEXT_TEST_ID = 'photo-card-text-test-id';
export const VIDEO_IFRAME_TEST_ID = 'video-iframe-test-id';
export const VIDEO_PLAY_BUTTON_TEST_ID = 'video-play-button-test-id';
export const VIDEO_TEST_ID = 'video-test-id';
export const VIDEO_THUMBNAIL_TEST_ID = 'video-thumbnail-test-id';
export const FLIP_CARD_BACK_TEST_ID = 'flip-card-back-test-id';
export const FLIP_CARD_FRONT_TEST_ID = 'flip-card-front-test-id';
export const FLIP_CARD_TEST_ID = 'flip-card-test-id';
export const SERVICE_CARD_FRONT_DESCRIPTION_TEST_ID =
  'service-card-front-description-test-id';
export const SERVICE_CARD_FRONT_HEADER_TEST_ID =
  'service-card-front-header-test-id';
export const SERVICE_CARD_BACK_HEADER_TEST_ID =
  'service-card-back-header-test-id';
export const SERVICE_CARD_BACK_LIST_ITEM_TEST_ID =
  'service-card-back-list-item-test-id';
export const SERVICE_CARD_BACK_LIST_TEST_ID = 'service-card-back-list-test-id';
export const SOCIAL_MEDIA_LABEL_TEST_ID = 'social-media-label-test-id';
export const SOCIAL_MEDIA_DESCRIPTION_TEST_ID =
  'social-media-description-test-id';
export const FOOTER_SECTION_TEST_ID = 'footer-section-test-id';
export const PAGINATION_DOT_TEST_ID = 'pagination-dot-test-id';
export const CAROUSEL_NAV_WRAPPER_TEST_ID = 'carousel-nav-wrapper-test-id';
export const CAROUSEL_NAV_NEXT_TEST_ID = 'carousel-nav-next-test-id';
export const CAROUSEL_NAV_PREV_TEST_ID = 'carousel-nav-prev-test-id';
export const ENDLESS_CAROUSEL_TEST_ID = 'endless-carousel-test-id';
export const ENDLESS_CAROUSEL_CONTENT_TEST_ID =
  'endless-carousel-content-test-id';
export const MAIN_SPRAY_BG_TEST_ID = 'main-spray-bg-test-id';
export const HERO_MAIN_CARD_ICON_TEST_ID = 'hero-main-card-icon-test-id';
export const HERO_TEXT_CONTENT_TEST_ID = 'hero-text-content-test-id';
export const HERO_TEXT_CONTENT_ICON_TEST_ID = 'hero-text-content-icon-test-id';
export const EXPERTISE_SECTION_TEST_ID = 'expertise-section-test-id';
export const SCROLL_UP_BUTTON_TEST_ID = 'scroll-up-button-test-id';
export const MODAL_WRAPPER_TEST_ID = 'modal-wrapper-test-id';
export const MODAL_CONTENT_TEST_ID = 'modal-content-test-id';
export const MODAL_CLOSE_BUTTON_TEST_ID = 'modal-close-button-test-id';
export const CASE_MODAL_TEXT_TEST_ID = 'case-modal-text-test-id';
export const TESTIMONIAL_MODAL_NAME_TEST_ID = 'testimonial-modal-name-test-id';
export const TESTIMONIAL_MODAL_POSITION_TEST_ID =
  'testimonial-modal-position-test-id';
export const TESTIMONIAL_MODAL_TEXT_TEST_ID = 'testimonial-modal-text-test-id';
export const TESTIMONIAL_MODAL_IMAGE_TEST_ID =
  'testimonial-modal-image-test-id';
