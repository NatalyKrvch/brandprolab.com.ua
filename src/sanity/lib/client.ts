import { createClient } from 'next-sanity';

import { getSanityConfig } from '../env';

const { apiVersion, dataset, projectId, useCdn } = getSanityConfig();

export const client = createClient({
  projectId,
  dataset,
  apiVersion,
  useCdn,
});
