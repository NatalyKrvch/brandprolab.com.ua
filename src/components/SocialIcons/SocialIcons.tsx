import Image from 'next/image';
import { twMerge } from 'tailwind-merge';

import { ExternalLink } from '@/components';
import { normalizeImageURL } from '@/utils';

import type { SocialIconsProps } from './types';

const SocialIcons = ({
  socialLinks,
  className,
  isColored,
}: SocialIconsProps) => {
  return (
    <ul
      aria-label="Соціальні мережі"
      className={twMerge('flex items-center gap-4', className)}
    >
      {socialLinks.map(link => {
        const src = isColored
          ? normalizeImageURL(link.colorIcon)
          : normalizeImageURL(link.whiteIcon);

        return (
          <li key={link._key}>
            <ExternalLink href={link.url}>
              <Image
                src={src}
                alt={`Посилання на ${link.label}`}
                width={32}
                height={32}
                priority
              />
            </ExternalLink>
          </li>
        );
      })}
    </ul>
  );
};

export default SocialIcons;
