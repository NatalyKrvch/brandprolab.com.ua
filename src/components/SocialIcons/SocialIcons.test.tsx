import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

jest.mock('yet-another-react-lightbox', () => {
  const MockLightbox = () => <div>Mock Lightbox</div>;
  MockLightbox.displayName = 'MockLightbox';
  return MockLightbox;
});
jest.mock('yet-another-react-lightbox/plugins/zoom', () => ({}));

import { EXTERNAL_LINK_TEST_ID } from '@/lib/testIDs';

import SocialIcons from './SocialIcons';
import type { SocialIconsProps } from './types';

const mkImg = (
  ref: string,
): {
  _type: 'image';
  asset: { _type: 'reference'; _ref: string };
} => ({
  _type: 'image',
  asset: { _type: 'reference', _ref: ref },
});

const mockSocialLinks: SocialIconsProps['socialLinks'] = [
  {
    _key: '1',
    _type: 'socialLink',
    label: 'Facebook',
    url: 'https://facebook.com',
    colorIcon: mkImg('color-123'),
    whiteIcon: mkImg('white-123'),
  },
  {
    _key: '2',
    _type: 'socialLink',
    label: 'Instagram',
    url: 'https://instagram.com',
    colorIcon: mkImg('color-456'),
    whiteIcon: mkImg('white-456'),
  },
];

jest.mock('next/image', () => {
  const MockImage = (props: any) => {
    // eslint-disable-next-line jsx-a11y/alt-text
    return <img {...props} />;
  };
  MockImage.displayName = 'MockNextImage';
  return {
    __esModule: true,
    default: MockImage,
  };
});

jest.mock('@/utils', () => ({
  ...jest.requireActual('@/utils'),
  normalizeImageURL: (image: any) =>
    image?.asset?._ref ? `/mocked-url-${image.asset._ref}` : '/mocked-fallback',
}));

jest.mock('@/sanity/lib/client', () => ({
  client: { fetch: jest.fn() },
}));

describe('SocialIcons component', () => {
  it('renders correct number of external links with proper attrs', () => {
    render(<SocialIcons socialLinks={mockSocialLinks} />);

    const links = screen.getAllByTestId(EXTERNAL_LINK_TEST_ID);
    expect(links).toHaveLength(mockSocialLinks.length);

    links.forEach((link, index) => {
      expect(link).toHaveAttribute('href', mockSocialLinks[index].url);
      expect(link).toHaveAttribute('target', '_blank');
      expect(link).toHaveAttribute('rel', 'noopener noreferrer');
    });
  });

  it('applies custom className to wrapper ul', () => {
    const { container } = render(
      <SocialIcons socialLinks={mockSocialLinks} className="custom-class" />,
    );
    const ul = container.querySelector('ul');
    expect(ul).toHaveClass('custom-class');
  });

  it('uses white icons by default (isColored undefined/false)', () => {
    render(<SocialIcons socialLinks={mockSocialLinks} />);

    const imgs = screen.getAllByRole('img');
    expect(imgs[0]).toHaveAttribute('src', '/mocked-url-white-123');
    expect(imgs[1]).toHaveAttribute('src', '/mocked-url-white-456');
  });

  it('uses colored icons when isColored=true', () => {
    render(<SocialIcons socialLinks={mockSocialLinks} isColored />);

    const imgs = screen.getAllByRole('img');
    expect(imgs[0]).toHaveAttribute('src', '/mocked-url-color-123');
    expect(imgs[1]).toHaveAttribute('src', '/mocked-url-color-456');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <SocialIcons socialLinks={mockSocialLinks} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
