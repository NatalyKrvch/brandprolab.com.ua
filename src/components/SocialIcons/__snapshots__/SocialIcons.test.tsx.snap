// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcons component matches snapshot 1`] = `
<DocumentFragment>
  <ul
    aria-label="Соціальні мережі"
    class="flex items-center gap-4"
  >
    <li>
      <a
        data-testid="external-link-test-id"
        href="https://facebook.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Посилання на Facebook"
          height="32"
          src="/mocked-url-white-123"
          width="32"
        />
      </a>
    </li>
    <li>
      <a
        data-testid="external-link-test-id"
        href="https://instagram.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Посилання на Instagram"
          height="32"
          src="/mocked-url-white-456"
          width="32"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
